codegen/templates/http_client.tmpl

```diff
@@ -240,6 +240,9 @@ func (c *{{$clientName}}) {{$methodName}}(
 	} else {
 		err = hystrix.DoC(ctx, "{{$clientID}}", func(ctx context.Context) error {
 			res, err = req.Do()
+			if res.StatusCode < 500 {
```

Comment:
The thought here is that if it is not a system error, we do not return failure to hystrix. But current code has one issue in that err gets overwritten.  If I update that line 241 above, that problem should go away.